pkg/apiserver/rest/dualwriter.go

```diff
@@ -222,10 +226,23 @@ func removeMeta(obj runtime.Object) []byte {
 		return nil
 	}
 	// we don't want to compare meta fields
-	delete(unstObj, "meta")
+	delete(unstObj, "metadata")
```

Comment:
I'm not 100% sure off the top of my head which fields go under the json "metadata" field, but we might want to include some of them in the comparison. For instance, the "Name" field tracks the UID and that would be important to check during the comparison. The name field is included under the "ObjectMeta field of the runtime object.

Reply:
hhm, but would that work with `update`? We override legacy's UID and RV but we don't override the name, so this will always be different, correct?

Reply:
name should be the same, otherwise they're different resources, that's the only field in metadata that we care about right now afaik, though we will care about folder annotation once that's implemented.

Reply:
So for now we could delete everything in the metadata except the name or do a comparison of the name in the Compare() function in addition to the comparison of the bytes.

Reply:
yeah, I think something like that could make sense. possibly just replace metadata with a new one that has only name from the old one, then it's easy to add more fields to the include list vs having to iterate and remove stuff